rust/src/storage/file/mod.rs

```diff
@@ -36,6 +36,13 @@ impl StorageBackend for FileStorageBackend {
             .unwrap()
     }
 
+    fn join_paths(&self, paths: &[&str]) -> String {
+        let mut iter = paths.iter();
+        let mut path = Path::new(iter.next().unwrap_or(&"")).to_path_buf();
```

Comment:
Maybe it is just me in the morning, but it took me a bit of time to understand this implementation.

What do you think about using PathBuf declaration directly (with from or a mut variable)?
`let mut path = PathBuf::new();`
`let path = PathBuf::from(string);`

Reply:
good suggestion 👍